src/ophys_etl/modules/segmentation/merge/characteristic_timeseries.py

```diff
@@ -0,0 +1,278 @@
+                                    args=args)
+        p.start()
+        process_list.append(p)
+        while len(process_list) > 0 and len(process_list) >= (n_processors-1):
+            process_list = _winnow_process_list(process_list)
+    for p in process_list:
+        p.join()
+    final_output = {}
+    k_list = list(output_dict.keys())
+    for k in k_list:
+        final_output[k] = output_dict.pop(k)
+    return final_output
+
+
+def update_timeseries_lookup(
+        merger_candidates: List[Tuple[int, int]],
+        timeseries_lookup: Dict[int, CharacteristicTimeseries],
+        sub_video_lookup: Dict[int, np.ndarray],
+        filter_fraction: float,
+        n_processors: int,
+        size_threshold: int = 500) -> Dict[int, CharacteristicTimeseries]:
+    """
+    Take a list of candidate merger ROI IDs and timeseries_lookup dict.
+    Update timeseries_lookup dict with the key pixel time series for
+    any pixels that are missing from the lookup table.
+
+    Parameters
+    ----------
+    merger_candidates: List[Tuple[int, int]]
+        A list of tuples representing potential ROI mergers
+
+    timeseries_lookup: Dict[int, dict]
+        A dict mapping ROI IDs to the characteristic timeseries
+        associated with those ROIs (see Notes for more details)
+
+    sub_video_lookup: Dict[int, np.ndarray]
+        A dict mapping ROI IDs to sub-videos which have been
+        flattened in space so that their shapes are (ntime, npixels)
+
+    filter_fraction: float
+        The fraction of timesteps (chosen to be the brightest) to
+        keep when doing the correlation.
+
+    n_processors: int
+        The number of processors to invoke with multiprocessing
+
+    size_threshold: int
+        The area at which an ROI gets parallelized at the pixel level
+        (default=500)
+
+    Returns
+    -------
+    timeseries_lookup: dict
+        Updated with any key pixels that need to be added.
+
+    Notes
+    -----
+    timeseries_lookup actually maps ROI IDs to another dict
+
+    timeseries_lookup[roi_id]['timeseries'] is the characteristic
+    time series of the ROI
+
+    timeseries_lookup[roi_id]['area'] is the area of the ROI
+    at the time when the characteristic time series was calculated
+    """
+
+    # if the ROIs are larger than 500 pixels in area,
+    # their characteristic time series will be calculated
+    # by a method that is parallelized at the pixel, rather
+    # than the ROI level
+    needed_big_rois = set()
+    needed_small_rois = set()
+
+    roi_to_consider = set()
+    for pair in merger_candidates:
+        roi_to_consider.add(pair[0])
+        roi_to_consider.add(pair[1])
+
+    for roi_id in roi_to_consider:
+        if roi_id not in timeseries_lookup:
+            area = sub_video_lookup[roi_id].shape[1]
+            if area >= size_threshold:
+                needed_big_rois.add(roi_id)
+            else:
+                needed_small_rois.add(roi_id)
+
+    new_small_pixels = {}
```

Comment:
`dict()` or `set()`

Reply:
done (they were all dicts; it never occurred to me that you can instantiate a set that way...)